gateway/ln-gateway/src/lib.rs

```diff
@@ -51,15 +73,16 @@ pub struct DepositPayload {
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct WithdrawPayload {
+    pub federation_id: FederationId,
     #[serde(with = "bitcoin::util::amount::serde::as_sat")]
     pub amount: bitcoin::Amount,
     pub address: Address,
 }
 
 #[derive(Debug)]
 pub enum GatewayRequest {
-    HtlcAccepted(GatewayRequestInner<HtlcAccepted>),
-    PayInvoice(GatewayRequestInner<ContractId>),
+    ReceiveInvoice(GatewayRequestInner<ReceiveInvoicePayload>),
```

Comment:
I don't think this is a good name. Maybe `IncomingHtlc` or something. There's no invoice involved here.

Reply:
`HtlcAccepted` translated to me like, "gateway receive invoice on behalf of user". I propose "ReceiveInvoice" on this basis, given the opposite direction is named "PayInvoice" which is nicely transparent.

Reply:
`HtlcAccepted` translated to me like, "gateway receive invoice on behalf of user". I propose "ReceiveInvoice" on this basis, given the opposite direction is named "PayInvoice" which is nicely transparent.

Reply:
`HtlcAccepted` translated to me like, "gateway receive invoice on behalf of user". I propose "ReceiveInvoice" on this basis, given the opposite direction is named "PayInvoice" which is nicely transparent.

Reply:
For the purposes of this PR the federation id can be user defined afaik? You could just give it any id on join.

While I do have a slight preference for making this id deterministically derived from the config and using this right away as `FederationId(Sha256)` instead of `FederationId(String)` I'd be ok with this for now if it unblocks you and accelerates resolution of #807 which is the highest priority on the GW rn imo.